src/app/js/admin/parsing/ParsingEditCell.js

```diff
@@ -28,71 +30,207 @@ const style = {
     },
     containedButton: {
         marginTop: 15,
-        float: 'right',
     },
     icon: {
         marginRight: 10,
     },
+    menuPaper: {
+        backgroundColor: theme.green.primary,
+        '& .MuiMenuItem-root': {
+            color: 'white',
+        },
+    },
 };
 const useStyles = makeStyles(style);
 
-const returnParsedValue = value => {
+export const returnParsedValue = value => {
     try {
         return JSON.parse(value);
     } catch (e) {
         return value;
     }
 };
 
-const isPrimitive = value => {
+export const isPrimitive = value => {
     return value !== Object(value);
 };
 
+export const tryParseJSONObjectOrArray = jsonString => {
+    try {
+        if (typeof jsonString === 'object') {
+            return jsonString;
+        }
+        const parsed = JSON.parse(jsonString);
+        if (parsed && (typeof parsed === 'object' || parsed instanceof Array)) {
+            return parsed;
+        }
+    } catch (e) {
+        return false;
+    }
+
+    return false;
+};
+
+export const getValueBySavingType = (value, type, previousValue) => {
+    if (type === 'number') {
+        const parsedValue = Number(value);
+        if (isNaN(parsedValue)) {
+            throw new Error('value_not_a_number');
+        }
+        return Number(value);
+    }
+    if (type === 'string') {
+        const parsedValue = tryParseJSONObjectOrArray(value);
+        if (parsedValue && typeof parsedValue === 'object') {
+            return JSON.stringify(value);
+        }
+        return String(value);
+    }
+    if (type === 'boolean') {
+        return value === 'true';
```

Comment:
what if we want to convert have number 1 equals to true ? In lodex doc about BOOLEAN transformer (https://lodex.inist.fr/docs/partie-2-2/transformer-des-donnees/#BOOLEAN) : 
Toute chaîne de caractères valant true, 1, on, ok, oui, yes ou true sera vraie, et toute autre chaîne sera fausse.